src/initialization/sensor_loader.py

```diff
@@ -55,45 +55,38 @@ def load_sensor(sensor_capabilities: dict) -> Sensor:
                 sensor_loc["y"],
                 sensor_loc["z"] if "z" in sensor_loc else None,
             )
-        switches = load_switches(settings.SWITCH_CONFIGS_DIR)
-        sensor_cal = get_sensor_calibration(
-            settings.SENSOR_CALIBRATION_FILE, settings.DEFAULT_CALIBRATION_FILE
-        )
-        sigan_cal = get_sigan_calibration(
-            settings.SIGAN_CALIBRATION_FILE, settings.DEFAULT_CALIBRATION_FILE
-        )
-        preselector = load_preselector(
-            settings.PRESELECTOR_CONFIG,
-            settings.PRESELECTOR_MODULE,
-            settings.PRESELECTOR_CLASS,
-            sensor_capabilities["sensor"],
-        )
 
     sigan = None
     try:
         if not settings.RUNNING_MIGRATIONS:
-            check_for_required_sigan_settings()
-            sigan_module_setting = settings.SIGAN_MODULE
-            sigan_module = importlib.import_module(sigan_module_setting)
-            logger.info(
-                "Creating " + settings.SIGAN_CLASS + " from " + settings.SIGAN_MODULE
-            )
-            sigan_constructor = getattr(sigan_module, settings.SIGAN_CLASS)
-            sigan = sigan_constructor(
-                sensor_cal=sensor_cal, sigan_cal=sigan_cal, switches=switches
-            )
-            register_component_with_status.send(sigan, component=sigan)
+            if get_usb_device_exists():
+                check_for_required_sigan_settings()
+                sigan_module_setting = settings.SIGAN_MODULE
+                sigan_module = importlib.import_module(sigan_module_setting)
+                logger.info(
+                    f"Creating {settings.SIGAN_CLASS} from {settings.SIGAN_MODULE}"
+                )
+                sigan_constructor = getattr(sigan_module, settings.SIGAN_CLASS)
+                sigan = sigan_constructor(switches=switches)
+                register_component_with_status.send(sigan, component=sigan)
+            else:
+                logger.warning("Required USB Device does not exist.")
         else:
             logger.info("Running migrations. Not loading signal analyzer.")
-    except Exception as ex:
+    except BaseException as ex:
         logger.warning(f"unable to create signal analyzer: {ex}")
+        set_container_unhealthy()
 
+    # Create sensor before handling calibrations
     sensor = Sensor(
         signal_analyzer=sigan,
+        # TODO GPS Not Implemented
```

Comment:
Should this be addressed now? SCOS Sensor in its current state already does ever pass a GPS to the sensor constructor. We may want to create a new issue to address this in the future

Reply:
@jhazentia

Reply:
Working on this in #272

Reply:
Sounds good, we will leave it alone for now.